src/rp2_common/hardware_rtc/rtc.c

```diff
@@ -131,43 +151,46 @@ static void rtc_irq_handler(void) {
     }
 }
 
-static bool rtc_alarm_repeats(datetime_t *t) {
-    // If any value is set to -1 then we don't match on that value
-    // hence the alarm will eventually repeat
-    if (t->year  < 0) return true;
-    if (t->month < 0) return true;
-    if (t->day   < 0) return true;
-    if (t->dotw  < 0) return true;
-    if (t->hour  < 0) return true;
-    if (t->min   < 0) return true;
-    if (t->sec   < 0) return true;
-    return false;
+// return != 0 on repeat, -1 on repeat every second
+static int8_t rtc_alarm_repeats(const datetime_t *t) {
```

Comment:
Should this be declared to return `repeat_type` rather than `int8_t`?

Reply:
done